src/components/CompanyList.js

```diff
@@ -0,0 +1,57 @@
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { Link } from 'react-router-dom';
+import '../styles/CompanyList.css';
+
+const CompanyList = ({ searchTerm }) => {
+  const { companies } = useSelector((state) => state.companies);
+
+  const filteredCompanies = companies.filter((company) => (
+    company.companyName.toLowerCase().includes(searchTerm.toLowerCase())
+       || company.symbol.toLowerCase().includes(searchTerm.toLowerCase())
+  ));
+  return (
+    <div className="company-wrapper">
+
+      {[...filteredCompanies]
+        .sort((a, b) => b.price - a.price)
+        .map((company, index) => (company.isLoading ? (
+          <div className="company-box" key={company.symbol}>
+            <span className="load-message">loading Company details ... </span>
+          </div>
+        )
+          : (
+            <>
+              <div className="company-box" key={company.symbol}>
+                <div className="box-top">
+                  <h3 className="rank">{index + 1}</h3>
+                  <Link to={`/Profile/${company.symbol}`}>
+                    <button type="button" className="back">
+                      more
+                      {' > '}
+                    </button>
+                  </Link>
+                </div>
+                <h3 className="company-name">{company.symbol}</h3>
+                <span className="stock-price">
+                  $
```

Comment:
- [ ] [OPTIONAL] I think it would be better that when you click on an item it opens the details for it instead of just clicking on more 
![image](https://user-images.githubusercontent.com/75126481/228993481-027cc52d-8ea0-4c2b-9af0-3028ab2574e5.png)

Reply:
Hi @hiromataba ,
Thank You for your review. Your suggestion really has weight and I would definitely improve the application based on that.

Thank You for approving this PR too 👌🏿